### PR TITLE
Modify the e2e local scripts so they can be run on a modern Mac

### PR DESCRIPTION
**Pull request by adrianmoisey**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

e2e tests don't run locally on a modern Apple Silicon Mac. Additionally, these scripts seem to be out of date.
This PR is an attempt to start updating the tooling for local dev and testing.
It also removes the registry container, as kind supports loading images directly, without the need for a registry.

I haven't looked at any failed tests yet, but I plan to make more follow up PRs to improve on this area of the VPA in the future.

#### Which issue(s) this PR fixes:

Related to https://github.com/kubernetes/autoscaler/issues/7455

#### Special notes for your reviewer:

Untested on Linux or amd64 machines.
See this PR as an attempt to improve on these tools, and not the final state, I'd like to clean it up more after this PR lands.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by omerap12 — COMMENTED** [review]
Could we upgrade kind version to version 1.30 or newer?
```
Deleting KIND cluster 'kind'.
Creating KIND cluster 'kind'
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.26.3) 🖼
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
```

**Comment by adrianmoisey** [comment]
> Could we upgrade kind version to version 1.30 or newer?

I was thinking about that, but decided to leave it. I want to look at other versions that we have in the project, and make another PR to bump them all to something reasonable. 

I kind of see this PR as "make it work" and a follow up PR will be "ensure everything is right", if you're fine with that

**Comment by omerap12** [comment]
> > Could we upgrade kind version to version 1.30 or newer?
> 
> I was thinking about that, but decided to leave it. I want to look at other versions that we have in the project, and make another PR to bump them all to something reasonable.
> 
> I kind of see this PR as "make it work" and a follow up PR will be "ensure everything is right", if you're fine with that

sounds good with me.

File: vertical-pod-autoscaler/hack/run-e2e-locally.sh
Comment (omerap12) [inline vertical-pod-autoscaler/hack/run-e2e-locally.sh:65]: How about setting it up like this for better readability:
```
KIND_VERSION="kindest/node:v1.26.3"
kind create cluster --image=${KIND_VERSION}
```

Comment (adrianmoisey) [inline vertical-pod-autoscaler/hack/run-e2e-locally.sh:65]:
```suggestion
KIND_VERSION="kindest/node:v1.26.3"
kind create cluster --image=${KIND_VERSION} 
```

**Comment by omerap12** [comment]
I tested it manually on a Mac, and it works.
/lgtm

**Comment by kwiesmueller** [comment]
/lgtm
/approve
/hold

Does anyone have a chance to run this on linux to verify those also still work?
Otherwise good to go, feel free to unhold once verified.

**Comment by omerap12** [comment]
> /lgtm /approve /hold
> 
> Does anyone have a chance to run this on linux to verify those also still work? Otherwise good to go, feel free to unhold once verified.

I can check this on a new ec2 over the weekend

**Comment by adrianmoisey** [comment]
I tested it on a fresh Linux VM, and it works.
/unhold

I found a few improvements to make, such as detecting if the necessary commands exist, such as go, kubectl and make.
I'll make a follow up PR for those.

**Comment by adrianmoisey** [comment]
@kwiesmueller I don't think the approve you did worked, can you re-approve?

**Comment by adrianmoisey** [comment]
Since the approve label wasn't applied, I made a few small improvements based on the testing on a fresh linux VM.

**Review by kwiesmueller — APPROVED** [review]
/lgtm
/approve
Thanks!

**Review by omerap12 — COMMENTED** [review]
Just a tiny comment otherwise lgtm from my side

File: vertical-pod-autoscaler/hack/run-e2e-locally.sh
Comment (omerap12) [reply vertical-pod-autoscaler/hack/run-e2e-locally.sh:54]: Great! Thanks for clarifying :)

**Review by omerap12 — APPROVED** [review]

**Comment by adrianmoisey** [comment]
Oh, the change to .gitignore requires a different approver.

@MaciekPytel can you take a look please?

/assign maciekpytel

**Comment by gjtempleton** [comment]
/approve
Thanks for this!

**Pull request merged**
